src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -544,24 +544,24 @@ private static bool TryParseVersion(string version, ref VersionResult result)
 
             var versionSansLabel = (dashIndex < 0) ? version : version.Substring(0, dashIndex);
             string[] parsedComponents = versionSansLabel.Split(Utils.Separators.Dot);
-            if (parsedComponents.Length != 3)
+            if (parsedComponents.Length == 0 || parsedComponents.Length > 3)
```

Comment:
This is C# - PowerShell ETS is not relevant, and you are inspecting the length of the first element, not the number of elements in the array. [msdn](https://msdn.microsoft.com/en-us/library/b873y76a(v=vs.110).aspx) does say:
```
If this instance does not contain any of the characters in separator, the returned array consists of a single element that contains this instance.
```